### DISCUSSION TITLE
Cant update a table

### DISCUSSION THREAD

**Discussion by TQuantab** [supabase/supabase#20675] [Questions] [answered]
Hi =)
im having a Problem with one of my tables, maybe someone can help =D

My table has the following definition:
```js
create table
  public.meeting_questions (
    id uuid not null default gen_random_uuid (),
    created_at timestamp with time zone not null default now(),
    meeting_id uuid not null,
    title text not null default ''::text,
    answer text not null default ''::text,
    block_id text not null,
    status public.question_status not null default 'open'::question_status,
    created_by uuid not null,
    constraint meeting_questions_pkey primary key (block_id),
    constraint meeting_questions_created_by_fkey foreign key (created_by) references profiles (id) on update cascade on delete cascade,
    constraint meeting_questions_meeting_id_fkey foreign key (meeting_id) references meetings (id) on update cascade on delete cascade
  ) tablespace pg_default;
  ```

For some reason, i cant update the table via supabase client. When doing it directly in supabases table editor, it works.
The table currently has no RLS, to make sure, my policies arent the Problem.

Im trying to update the table like this:
```js
function updateOnButtonPress() {
	supabase
		.from('meeting_questions')
		.update('title', 'Test123456')
		.eq('id', '8fc179d3-6a37-4e69-bc91-a96d22c8493c')
		.select()
		.then(({ data, error }) => {
			if (error) {
				console.log(error);
			} else {
				console.log(data);
				console.log(question.id);
				console.log('updated Question');
				invalidate('supabase:question');
			}
		});
}
```

This does not work, and doesnt even give me an error :/ It seems to just silently fail.

Does anyone have an Idea?

**Comment by GaryAustin1** [comment]
When you say "has no RLS" do you mean no policies or it is disabled for the table.?
Silent return on update is usually not meeting RLS or not meeting your filter.

**Reply by TQuantab** [reply]
I have policies, which work (for select and update). But i disabled RLS entirely, to make sure, they are not the Problem.
Interestingly, updating the questions status works fine.

**Reply by TQuantab** [reply, marked as answer]
I Fixed it, i forgot the "{}" in the update clause

**Comment by busyaperu** [comment]
Hello, can you help me, I have a problem using DELETE, the RLS is as follows: (SELECT (auth.email() = accredit.email)), when I do a test with Postman it returns 204, but the row is not deleted. I'm doing wrong. Thanks for your help

This is incredible, everything works fine but DELETE DOES NOT WORK and there is no where to find a solution, this can harm SUPABASE

**Reply by GaryAustin1** [reply]
Do you meet both select and delete RLS?
RLS acts like the rows do not exists and is silent if you don't meet it.

**Comment by busyaperu** [comment]
Hi Gary

Thank you very much for your support, I don't understand what you mean by:
Does it comply with selecting and deleting RLS?
The RLS is as follows: (SELECT (auth.email() = accredit.email)), when I do
a test with Postman it returns 204, but the row is not deleted, what am I
doing wrong? First I want to try Postman to make sure DELETE works.

I wait for you support

Regards

Jorge Camus

**Reply by GaryAustin1** [reply]
If you set that policy to true does it work?  Can you show your SELECT and DELETE (or ALL) policies.  Both have to be met.

What jwt are you passing in the Authorization header when you test from Postman?

**Comment by busyaperu** [comment]
Gary

Estas son las políticas que tome del template de Supabase:
SELECT policies

create policy "Public acreditar are visible to everyone."
on acreditar for select
to anon         -- the Postgres Role (recommended)
using ( true ); -- the actual Policy

DELETE policies

create policy "Users can delete a acreditar."
on acreditar for delete
to authenticated
 -- the Postgres Role (recommended)
using ( SELECT (auth.email() = acreditar.email)) ;      -- the actual Policy

El encabezado de Autorización cuando pruebas desde Postman, es el "apikey"

Saludos
Jorge Camus

**Reply by GaryAustin1** [reply]
Your second policy requires you have a user token(jwt) in the Authorization header as 'Bearer jwt'.
You need both an apikey with the anon jwt token AND an Authorization header with a user jwt.
Auth.email() is looking in the jwt to see if that email is in there.
If you change that policy to TO anon and the USING to true.  See if it works.

**Reply by GaryAustin1** [reply]
I don't speak Spanish.  I glanced at the video and this image shows an authorization header.  I don't know where it came from or what it is set to.  
![image](https://github.com/supabase/supabase/assets/54564956/03610818-7057-4a84-bf71-21a15d00f45c)

But the policy you shows requires a USER jwt that is not expired and contains a claim called email.  A normal signed in user on Supabase will have that in the jwt.  

Did you change the policy (anon/true) like I suggested to see if it runs?

You can't use auth.uid() or auth.email() without a user jwt.  It will not work.

**Reply by GaryAustin1** [reply]
The line underlines the Authorization header which to use your RLS is a user jwt after Bearer.  If you put anon key there then auth.uid() or auth.email() won't work.

Also I'm having to run your comment thru Google translate so have no idea if it is correct or not.

**Comment by busyaperu** [comment]
Gary

Parece que según este video en el minuto 28:15 , el DELETE es muy sencillo
y las RLS también son sencillas , no requiere JWT además está en public.
Realiza pruebas con Postman y funciona muy bien. Entonces quizá este video
pueda dar una idea de que estoy haciendo mal.

https://www.youtube.com/watch?v=pi33WDrgfpI

Quedo a los comentarios

Saludos

Jorge

**Comment by busyaperu** [comment]
Gary
La imagen donde señalas con una linea roja es el "apikey" es decir el
"barer apikey" , no hace mencion a jwt, por otro lado (anon / true) sale
mensaje de error cuando quiero guardar la RLS.

Saludos
Jorge
